### PR TITLE
Add flop formulas for sdpa

### PR DESCRIPTION
**Pull request by Chillee**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #95940
* #95901
* __->__ #95854
* #95751
* #95355

### CODE REVIEW

**Review by albanD — COMMENTED** [review]
cc @drisspg

File: torch/utils/flop_counter.py [hunk 1]
```diff
@@ -119,6 +119,29 @@ def conv_backward_flop(input_shapes: List[Any], output_shapes: List[Any]) -> int
...
+    Count flops for self-attention.
+    """
+    b, h, s, d = query_shape
+    b2, h2, s2, d2 = key_shape
+    assert b == b2 and h == h2 and d == d2
+    total_flops = 0
+    # [b, h, s, d] @ [b, h, d, s2] -> [b, h, s, s2]
+    total_flops += bmm_flop([(b * h, s, d), (b * h, d, s2)])
+    # [b, h, s, s2] @ [b, h, s2, d] -> [b, h, s2, d]
+    total_flops += bmm_flop([(b * h, s, s2), (b * h, s2, d)])
```
Comment (drisspg) [inline torch/utils/flop_counter.py:133]: Not sure how precise flop_counting is but in essence, we only really care about the bmms right?

**Review by drisspg — APPROVED** [review]

File: torch/utils/flop_counter.py (hunk 1 above)
Comment (Chillee) [reply torch/utils/flop_counter.py:133]: more or less yeah.

**Comment by Chillee** [comment]
I still need to add the backwards formulas, but I'll leave that to a future PR 🤔

**Pull request closed without merging**